.github/workflows/build-and-test.yml

```diff
@@ -130,6 +130,31 @@ jobs:
             echo "One or more matrix jobs failed."
             false
           fi
+  govulncheck:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ~1.19.7
+      - name: Cache Go
+        id: go-cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/bin
+            ~/go/pkg/mod
+          key: go-cache-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
+      - name: Install Tools
+        if: steps.go-cache.outputs.cache-hit != 'true'
+        run: make install-tools
+      - name: Run `govulncheck`
+        run: |
+          make govulncheck
+          git diff --exit-code || (echo 'Some vulnerable dependencies are in use, this most likely means you need to updated stuff in go.mod/go.sum and/or bump the Go compiler version itself.' && exit 1)
```

Comment:
Approved too soon. I just tried the command, and it looks like it doesn't change the codebase but exits with an error code and a descriptive message about found vulnerabilities. So `git diff` is not needed here. We need to make sure the job is failing if the exist code != 0

Reply:
Ahh, good catch, thanks! What a silly copy-paste mistake... I'll fix it early next week when I'm back from my Easter break :)